lib/pengine/unpack.c

```diff
@@ -1910,17 +1914,31 @@ process_rsc_state(resource_t * rsc, node_t * node,
             break;
         case action_fail_reset_remote:
             set_bit(rsc->flags, pe_rsc_failed);
-            tmpnode = NULL;
-            if (rsc->is_remote_node) {
-                tmpnode = pe_find_node(data_set->nodes, rsc->id);
+            if (is_set(data_set->flags, pe_flag_stonith_enabled)) {
+                tmpnode = NULL;
+                if (rsc->is_remote_node) {
+                    tmpnode = pe_find_node(data_set->nodes, rsc->id);
+                }
+                if (tmpnode &&
+                    is_baremetal_remote_node(tmpnode) &&
+                    tmpnode->details->remote_was_fenced == 0) {
```

Comment:
I don't think the timestamp of last_failure and fencing are related.  A "monitor" op may have triggered fencing, but perhaps the next last_failure is "stop" because the cluster node that was hosting the remote node connection is hosed... (when monitor fails, stop is still executed on the connection even though the remote node is fenced)  maybe this would make it look like the remote_node needs to get fenced again when it fact it's already been fenced?

I'm also a little nervous about comparing timestamps from two separate nodes. when we actually look at timestamps to mean something, all the instances I can think of involve us comparing timestamps that originated from the same node.  this isn't the case with fencing. the monitor op could fail on node1, fencing could occur on node2, if the node's aren't synced perfectly we could  be opening up a timing gap.

When 'was_fenced' is set != 0, that means the remote node has been fenced, and that no cluster node has successfully reconnected to the remote node since that fencing operation took place. There shouldn't be any timing issues. successfully fencing results in immediately writing the value in the fencing callback, a successful start action immediately clears the value in the start action callback.